tools/lint.py

```diff
@@ -4,29 +4,29 @@
 
 import os
 import sys
-from util import enable_ansi_colors, find_exts, root_path, run
+from util import enable_ansi_colors, find_exts, root_path, run, third_party_path
 from third_party import python_env
 
 enable_ansi_colors()
 
-third_party_path = os.path.join(root_path, "third_party")
-cpplint = os.path.join(third_party_path, "cpplint", "cpplint.py")
-eslint = os.path.join(third_party_path, "node_modules", "eslint", "bin",
-                      "eslint")
+cpplint_script = os.path.join(third_party_path, "cpplint", "cpplint.py")
+eslint_script = os.path.join(third_party_path, "node_modules", "eslint", "bin",
+                             "eslint")
+pylint_script = os.path.join(third_party_path, "python_packages", "pylint")
+pylint_rcfile = os.path.join(third_party_path, "depot_tools", "pylintrc")
 
 os.chdir(root_path)
+
 run([
-    "python", cpplint, "--filter=-build/include_subdir",
+    sys.executable, cpplint_script, "--filter=-build/include_subdir",
     "--repository=core/libdeno", "--extensions=cc,h", "--recursive", "core"
 ])
 
 run([
-    "node", eslint, "--max-warnings=0", "./js/**/*.{ts,js}",
+    "node", eslint_script, "--max-warnings=0", "./js/**/*.{ts,js}",
     "./core/**/*.{ts,js}", "./tests/**/*.{ts,js}"
 ])
 
-run([
-    sys.executable, "third_party/python_packages/pylint",
-    "--rcfile=third_party/depot_tools/pylintrc"
-] + find_exts(["tools", "build_extra"], [".py"], skip=["tools/clang"]),
+run([sys.executable, pylint_script, "--rcfile=" + pylint_rcfile] + find_exts(
+    ["tools", "build_extra"], [".py"], skip=["tools/clang"]),
```

Comment:
This lint rule wasn't updated after the `tools/build.py` removal and snapshot build refactor and should be
`["tools", "deno_typescript", "cli_snapshots"], [".py"], skip=["tools/clang"]),` also the `build_extra` for yapf in `tools/format.py` is now superfluous too.

(and this fix would fit better here than in #2943)

Reply:
Also, I suggest adding a new utility function 
```py
def find_exts2(dirs, exts):
    skip_dirs = ["tools/clang", "js/deps"]
    return find_exts(dirs, exts, skip=skip_dirs)
```
Because the directories that we skip is constant and should be used for all lint and format calls to find_exts.

Reply:
You can leave out the "deno_typescript" and "cli_snapshots" directories too - they should not contain any python.

Please remove [this dead code](https://github.com/denoland/deno/blob/master/cli_snapshots/run.py) too.

Reply:
I'm going to do it completely differently. Which is: run `git --ls-files <flags>` to get a list of files, which does include untracked files, and automatically excludes submodules. With that we don't have to be so precise about including/excluding the right directories. Additionally it's also faster and less brittle.

Reply:
>Because the directories that we skip is constant and should be used for all lint and format calls to find_exts. 

Looks like a good idea and once we move the gn root to `core/libdeno` we just change this from `"tools/clang"` to ignoring all the symlinks in "tools/clang" explicitly.

Reply:
SGTM but on CI it should still check everything

Reply:
^--- Note this requirement means that we still need to explicitly list the dirs in the code. I would put off `git --ls-files` until later because it's strictly more code and only an optimization.

Reply:
> ^--- Note this requirement means that we still need to explicitly list the dirs in the code. I would put off `git --ls-files` until later because it's strictly more code and only an optimization.

I don't understand what you're saying. You want to keep it like it is today, where half of the files are not linted? (https://github.com/denoland/deno/pull/2950/files#diff-377afa92851e34fdf91413c0102c08fbR42-R43)

The purpose of this change is to make it to (mostly) automatically lint/format the right files instead of manually including/excluding them.

Reply:
Sorry, I misunderstood the usage of `git ls-files`. I thought it only looked at the modified files. This works!